mypy/semanal_enum.py

```diff
@@ -104,23 +104,37 @@ def parse_enum_call_args(self, call: CallExpr,
         Return a tuple of fields, values, was there an error.
         """
         args = call.args
+        if not all([arg_kind in [ARG_POS, ARG_NAMED] for arg_kind in call.arg_kinds]):
+            return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
         if len(args) < 2:
             return self.fail_enum_call_arg("Too few arguments for %s()" % class_name, call)
-        if len(args) > 2:
+        if len(args) > 6:
             return self.fail_enum_call_arg("Too many arguments for %s()" % class_name, call)
-        if call.arg_kinds != [ARG_POS, ARG_POS]:
-            return self.fail_enum_call_arg("Unexpected arguments to %s()" % class_name, call)
-        if not isinstance(args[0], (StrExpr, UnicodeExpr)):
+        valid_name = [None, 'value', 'names', 'module', 'qualname', 'type', 'start']
+        for arg_name in call.arg_names:
+            if arg_name not in valid_name:
+                self.fail_enum_call_arg("Unexpected keyword argument '{}'".format(arg_name), call)
+        value, names = None, None
+        for arg_name, arg in zip(call.arg_names, args):
+            if arg_name == 'value':
+                value = arg
+            if arg_name == 'names':
+                names = arg
```

Comment:
These additional improvements would be nice: 

* Make sure that only at most the first two arguments are given as positional arguments (the others are keyword-only).
* If some argument is given as a keyword argument, ensure that they aren't also given as positional arguments.